lukefi/metsi/forestry/cross_cutting/stem_profile.py

```diff
@@ -1,88 +1,77 @@
         t1 = 1.0814409
         t4 = 0.8409653
         t7 = 0.4999158
-        #
-        y1 = (-0.003133*d + 0.01172*h + 0.48952*dh - 0.078688*dh2 -
-        0.31296*dl + 0.13242*hl - 1.2967/h)
-        y4 = (-0.0065534*d + 0.011587*h - 0.054213*dh + 0.011557*dh2 +
-        0.12598/h)
-        y7 = (0.084893 - 0.0064871*d + 0.012711*h - 0.10287*dh +
-        0.026841*dh2 - 0.01932*dl)
-
-    if sp=="birch":
+
+        y1 = -0.003133*d + 0.01172*h + 0.48952*dh - 0.078688*dh2 - 0.31296*dl + 0.13242*hl - 1.2967/h
+        y4 = -0.0065534*d + 0.011587*h - 0.054213*dh + 0.011557*dh2 + 0.12598/h
+        y7 = 0.084893 - 0.0064871*d + 0.012711*h - 0.10287*dh + 0.026841*dh2 - 0.01932*dl
+
+    elif sp == 3:  # birch
         t1 = 1.084544
         t4 = 0.8417135
         t7 = 0.4577622
-        
-        y1 = (0.59848 + 0.011356*d - 0.49612*dl + 0.46137*hl -
-            0.92116/dh + 0.25182/dh2 - 0.00019947*d2)
-        y4 = (-0.96443 + 0.011401*d + 0.13870*dl + 1.5003/h +
-            0.57278/dh - 0.18735/dh2 - 0.00026*d2)
-        y7 = (-2.1147 + 0.79368*dl - 0.51810*hl + 2.9061/h +
-            1.6811/dh - 0.40778/dh2 - 0.00011148*d2)
-    
-    if sp=="alnus":
+
+        y1 = 0.59848 + 0.011356*d - 0.49612*dl + 0.46137*hl - 0.92116/dh + 0.25182/dh2 - 0.00019947*d2
+        y4 = -0.96443 + 0.011401*d + 0.13870*dl + 1.5003/h + 0.57278/dh - 0.18735/dh2 - 0.00026*d2
+        y7 = -2.1147 + 0.79368*dl - 0.51810*hl + 2.9061/h + 1.6811/dh - 0.40778/dh2 - 0.00011148*d2
+
+    elif sp == 4:  # alnus
         t1 = 1.108743
         t4 = 0.8186044
         t7 = 0.4682397
-        #
-        y1 = (-1.46153 + 0.0487415*d + 0.663667*dl - 0.827114*hl -
-        0.00106612*d2 + 1.87966/h + 1.85706/dh - 0.467842/dh2)
-        y4 = (-1.24788 - 0.0218693*dh2 + 0.496483*dl - 0.291413*hl +
-        1.92579/h + 0.863274/dh - 0.183220/dh2)
-        y7 = (-0.478730 - 0.104679*dh + 0.151028*dl + 0.882010/h +
-        0.178386/dh)
 
-    y1t = min(abs(y1),0.1)
+        y1 = -1.46153 + 0.0487415*d + 0.663667*dl - 0.827114*hl - 0.00106612*d2 + 1.87966/h + 1.85706/dh - 0.467842/dh2
+        y4 = -1.24788 - 0.0218693*dh2 + 0.496483*dl - 0.291413*hl + 1.92579/h + 0.863274/dh - 0.183220/dh2
+        y7 = -0.478730 - 0.104679*dh + 0.151028*dl + 0.882010/h + 0.178386/dh
 
-    if np.sign(y1t) != np.sign(y1):
-        y1t = y1t * -1
+    # Clamp and preserve sign
+    def clamp(y):
+        y_clamp = min(abs(y), 0.1)
+        if y_clamp * y < 0:
+            y_clamp = -y_clamp
+        return y_clamp
 
-    y4t = min(abs(y4),0.1)
-
-    if np.sign(y4t) != np.sign(y4):
-        y4t = y4t * -1
-
-    y7t = min(abs(y7),0.1)
-
-    if np.sign(y7t)!=np.sign(y7):
-        y7t = y7t * -1
+    y1t = clamp(y1)
+    y4t = clamp(y4)
+    y7t = clamp(y7)
 
     p = np.zeros(5)
     p[0] = 0.9
     p[1] = 0.6
-    p[2] = t1/(t1+y1) * (t4+y4) - t4
+    p[2] = t1 / (t1 + y1t) * (t4 + y4t) - t4
     p[3] = 0.3
-    p[4] = t1/(t1+y1) * (t7+y7) - t7
+    p[4] = t1 / (t1 + y1t) * (t7 + y7t) - t7
```

Comment:
`y1t `and `y7t` have not been in use before. Let's validate this by the original author of the code.

Reply:
This was approved by original author.